src/commands/DownloadCommand.ts

```diff
@@ -22,35 +25,49 @@ export class DownloadCommand {
    * @param {string} outputFolder
    * @memberof DownloadCommand
    */
-  constructor(private originOrganization: string, originApiKey: string, private outputFolder: string) {
-    this.oganization = new Organization(originOrganization, originApiKey);
+  constructor(originOrganization: string, originApiKey: string, private outputFolder: string) {
+    this.organization = new Organization(originOrganization, originApiKey);
   }
 
   /**
    * Downloads the fields of organization. Produces a ${objectName}.json file
    * @memberof DownloadCommand
    */
   downloadFields() {
-    // hack: FieldController needs 2 orgs becaase initially it was  meant for comparison. Feed it.
-    const dummyOrg = new Organization('dummy', 'dummy');
-    const fieldController: FieldController = new FieldController(this.oganization, dummyOrg);
+    const fieldController: FieldController = new FieldController(this.organization);
     this.download(fieldController, 'Field', (field: Field) => field.getConfiguration());
   }
 
+  /**
+   * Downloads the sources of organization. Produces a ${objectName}.json file
+   * @memberof DownloadCommand
+   */
+  downloadSources() {
+    const sourceController: SourceController = new SourceController(this.organization);
+    this.download(sourceController, 'Source', (source: Source) => source.getConfiguration());
+  }
+
+  /**
+   * Downloads the extensions of organization. Produces a ${objectName}.json file
+   * @memberof DownloadCommand
+   */
+  downloadExtensions() {
+    const extensionController: ExtensionController = new ExtensionController(this.organization);
+    this.download(extensionController, 'Extension', (extension: Extension) => extension.getConfiguration());
+  }
+
   private download(controller: BaseController, objectName: string, extractionMethod: (object: any) => IStringMap<any>) {
-    Logger.startSpinner('Downloading fields');
+    Logger.startSpinner(`Downloading ${objectName}s`);
 
     controller
-      .download(this.originOrganization)
-      .then((downloadResultArray: IDownloadResultArray) => {
+      .download()
+      .then((downloadResultArray: DownloadResultArray) => {
         // sort the items in the list
         downloadResultArray.sort();
+
         // get the list
-        const items = downloadResultArray.getItems();
-        // sort items' attributes
-        items.forEach((value: BaseCoveoObject, index: number, array: BaseCoveoObject[]) => {
```

Comment:
@jpdery26 I removed this section because it was changing the nature of some objects if it contained arrays.

This method is also used to download sources and extensions, so it could have caused issues whenever the object did contain an array.

I guess we can update the Field extraction method to reproduce that behaviour.